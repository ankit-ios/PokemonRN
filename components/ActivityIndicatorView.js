import { ActivityIndicator, StyleSheet } from "react-native";

const ActivityIndicatorView = ({
  size = "large",
  color = "#fff",
  style,
}) => {
  return (
    <ActivityIndicator
      style={[styles.activityIndicator, style]}
      size={size}
      color={color}
    />
  );
};

const styles = StyleSheet.create({
  activityIndicator: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
  },
});

export default ActivityIndicatorView;
