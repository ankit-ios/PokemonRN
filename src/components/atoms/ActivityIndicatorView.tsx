import React from "react";
import { ActivityIndicator, StyleSheet } from "react-native";

interface ActivityIndicatorProps {
  size?: number | "small" | "large" | undefined;
  color?: string;
  style?: any;
}

const ActivityIndicatorView = ({
  size = "large",
  color = "#fff",
  style = [],
}: ActivityIndicatorProps) => {
  return (
    <ActivityIndicator
      testID="activity-indicator"
      style={[styles.activityIndicator, style]}
      size={size}
      color={color}
    />
  );
};

const styles = StyleSheet.create({
  activityIndicator: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
  },
});

export default ActivityIndicatorView;
