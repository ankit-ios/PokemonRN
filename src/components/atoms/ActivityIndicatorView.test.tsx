import React from 'react';
import { render } from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer'
import ActivityIndicatorView from './ActivityIndicatorView';

describe('ActivityIndicatorView', () => {
  it('renders correctly with default props', () => {
    const { getByTestId } = render(<ActivityIndicatorView />);
    const activityIndicator = getByTestId('activity-indicator');
    
    expect(activityIndicator.props.size).toBe('large');
    expect(activityIndicator.props.color).toBe('#fff');
  });

  it('applies custom size and color props', () => {
    const { getByTestId } = render(<ActivityIndicatorView size="small" color="#000" />);
    const activityIndicator = getByTestId('activity-indicator');
    
    expect(activityIndicator.props.size).toBe('small');
    expect(activityIndicator.props.color).toBe('#000');
  });

  it('merges custom style with default style', () => {
    const customStyle = { backgroundColor: 'red' };
    const { getByTestId } = render(<ActivityIndicatorView style={customStyle} />);
    const activityIndicator = getByTestId('activity-indicator');
    
    expect(activityIndicator.props.style).toContainEqual(customStyle);
    expect(activityIndicator.props.style).toContainEqual({
      position: "absolute",
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      alignItems: "center",
      justifyContent: "center",
    });
  });

  it('renders correctly', () => {
    const tree = TestRenderer.create(<ActivityIndicatorView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
