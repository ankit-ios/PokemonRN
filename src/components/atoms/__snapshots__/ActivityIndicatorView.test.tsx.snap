// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicatorView renders correctly 1`] = `
<ActivityIndicator
  color="#fff"
  size="large"
  style={
    [
      {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      [],
    ]
  }
  testID="activity-indicator"
/>
`;
